client/client.go

```diff
@@ -102,7 +109,38 @@ func NewApiClient(conf Config) (*ApiClient, error) {
 	return New(conf)
 }
 
-// Client initialization
+// keepaliveDialer is used to set keepalive options on the socket used for
+// connection. This is for handling connection timeouts while server
+// is not accessible after connection was alrady started.
+type keepaliveDialer struct {
+	net.Dialer
+}
```

Comment:
Indeed there is a `KeepAlive` field which could do the job. The only drawback is that it is quite limited. You can specify the only one interval which is used both as `keepalice_time` and `keepalive_interval`. What is more, currently there is no way to specify `keepalive_probes`, which is obviously impacting the whole connection timeout.
For the sake of simplicity, MAYBE, we can use that.